quic/s2n-quic-transport/src/path/manager/tests.rs

```diff
@@ -970,6 +970,130 @@ fn limit_number_of_connection_migrations() {
     assert_eq!(total_paths, MAX_ALLOWED_PATHS);
 }
 
+#[test]
+fn active_connection_migration_disabled() {
+    // Setup:
+    let mut publisher = Publisher::snapshot();
+    let new_addr: SocketAddr = "127.0.0.1:1".parse().unwrap();
+    let new_addr = SocketAddress::from(new_addr);
+    let new_addr = RemoteAddress::from(new_addr);
+    let first_path = ServerPath::new(
+        new_addr,
+        connection::PeerId::try_from_bytes(&[1]).unwrap(),
+        connection::LocalId::TEST_ID,
+        RttEstimator::default(),
+        Default::default(),
+        false,
+        mtu::Config::default(),
+    );
+    let mut manager = manager_server(first_path);
+    // Give the path manager soms new CIDs so it's able to use one for an active migration
+    let id_2 = connection::PeerId::try_from_bytes(b"id02").unwrap();
+    assert!(manager
+        .on_new_connection_id(&id_2, 1, 0, &TEST_TOKEN_1, &mut publisher)
+        .is_ok());
+    let id_3 = connection::PeerId::try_from_bytes(b"id03").unwrap();
+    assert!(manager
+        .on_new_connection_id(&id_3, 2, 0, &TEST_TOKEN_2, &mut publisher)
+        .is_ok());
+
+    let new_addr: SocketAddr = "127.0.0.2:1".parse().unwrap();
+    let new_addr = SocketAddress::from(new_addr);
+    let new_addr = RemoteAddress::from(new_addr);
+    let new_cid = connection::LocalId::try_from_bytes(b"id02").unwrap();
+    let now = NoopClock {}.get_time();
+    let datagram = DatagramInfo {
+        timestamp: now,
+        payload_len: 0,
+        ecn: ExplicitCongestionNotification::default(),
+        destination_connection_id: new_cid,
+        destination_connection_id_classification: connection::id::Classification::Local,
+        source_connection_id: None,
+    };
+
+    // Active connection migration is disabled
+    let limits = Limits::default().with_connection_migration(false).unwrap();
+
+    let res = manager.handle_connection_migration(
+        &new_addr,
+        &datagram,
+        &mut Default::default(),
+        &mut migration::allow_all::Validator,
+        mtu::Config::default(),
+        &limits,
+        &mut publisher,
+    );
+
+    assert!(matches!(
+        res,
+        Err(DatagramDropReason::RejectedConnectionMigration)
+    ));
+    assert_eq!(1, manager.paths.len());
+
+    // Active connection migration is enabled (default)
+    let res = manager.handle_connection_migration(
+        &new_addr,
+        &datagram,
+        &mut Default::default(),
+        &mut migration::allow_all::Validator,
+        mtu::Config::default(),
+        &Limits::default(),
+        &mut publisher,
+    );
+
+    assert!(res.is_ok());
+    assert_eq!(2, manager.paths.len());
+}
+
+#[test]
+fn active_connection_migration_disabled_passive_migration() {
```

Comment:
The test name might be a little difficult to follow.. maybe a comment?
- active conn migration is disabled
- we attempt passive migration and ensure it succeeds

Reply:
For the two test cases is it possible to extract the setup into a common fn by itself?

Reply:
I'll combine the two tests

Reply:
added some more comments too